### ISSUE TITLE
🚀 Refactor the Codebase for Scalability

### ISSUE THREAD

**Issue by Parnab03** [Parnab03/WallGodds#7]
**Description**: We need to reorganize the current code to make it more modular. This means breaking the code into smaller, reusable parts. Right now, the Sidebar module is included inside the Navbar module. To make it easier to manage, we should move the Sidebar module to be part of the Gallery module instead. This will fix several issues and improve the structure.
**Skills Required:** React, JavaScript, Best Practices.
**Deliverable:** Cleaner, modular, and scalable code.

**Comment by Lonwwolf14** [comment]
Hii @Parnab03 I would like to work on this Issues for refactoring the codebase as a part of WOC 4.0

**Comment by Parnab03** [comment]
@Lonwwolf14  

We’re excited to assign this issue to you! It’s a crucial one for the entire project, and we’ll be here to guide you if you face any challenges. If you have any queries, feel free to ask in the Discord Frontend Discussion channel.  

Looking forward to seeing your contributions!

**Comment by Lonwwolf14** [comment]
I need to solve the issues mentioned in the above description first, right?
Then focusing on the Whole project's codebase

**Comment by Parnab03** [comment]
@Lonwwolf14  

Yes, the issue you're working on is key to fixing the gallery module, which is our current focus. It can be resolved by addressing the problem as described in the issue description.

**Comment by dhrupatel031205** [comment]
i would like to work on this issue to also imrove my skills on the reusability and also help to the community

**Comment by Parnab03** [comment]
@dhrupatel031205 

We’re excited to assign this issue to you! It’s a crucial one for the entire project, and we’ll be here to guide you if you face any challenges. If you have any queries, feel free to ask in the Discord Frontend Discussion channel.

Looking forward to seeing your contributions!

**Comment by dhrupatel031205** [comment]
If I have any doubt related and I was stuck anywhere can I tell you here or I need to chat on discord server ...?

**Comment by Parnab03** [comment]
> If I have any doubt related and I was stuck anywhere can I tell you here or I need to chat on discord server ...?

chat on discord server is always better than here.

**Comment by dhrupatel031205** [comment]
ok sir
